mooncake-store/include/master_service.h

```diff
@@ -214,8 +214,10 @@ class MasterService {
     std::thread gc_thread_;
     std::atomic<bool> gc_running_{false};
     bool enable_gc_{true};  // Flag to enable/disable garbage collection
+    bool need_eviction_{false}; // Set to trigger eviction when not enough space left
```

Comment:
Should we be using `atomic<bool>` here? Seems like we've got some concurrent read/write action going on.

Reply:
sounds great